ledger/ledgercore/statedelta.go

```diff
@@ -160,19 +157,19 @@ type AssetResourceRecord struct {
 // The map would point the address/address+creatable id onto the index of the
 // element within the slice.
 type AccountDeltas struct {
```

Comment:
@algorandskiy converting to a thread

> How's the StateDelta is used (which methods are needed) ? I guess only `GetByIdx`, `Len` and `GetResource` are needed. Only the last one requires maps for lookup.

When the object has been created by the ledger, we only need the public methods. This is all code that Tolic and Tsachi wrote. The entry point is here: https://github.com/algorand/indexer/blob/develop/idb/postgres/internal/writer/writer.go#L327

It's split out pretty nicely in [writeAccountDeltas](https://github.com/algorand/indexer/blob/f176b4f607a28231f6d71e90129bdb9ec228582e/idb/postgres/internal/writer/writer.go#L268):
* [writeAccount(... ledgercore.AccountData ...)](https://github.com/algorand/indexer/blob/f176b4f607a28231f6d71e90129bdb9ec228582e/idb/postgres/internal/writer/writer.go#L182)
* [writeAppResource(... ledgercore.AppResourceRecord ...)](https://github.com/algorand/indexer/blob/f176b4f607a28231f6d71e90129bdb9ec228582e/idb/postgres/internal/writer/writer.go#L246)
* [writeAssetResource(... ledgercore.AssetResourceRecord ...)](https://github.com/algorand/indexer/blob/f176b4f607a28231f6d71e90129bdb9ec228582e/idb/postgres/internal/writer/writer.go#L223)
* A similar method for boxes has been written.

Reply:
Right, so only `accts`, `appResources`, `assetResources` should be exported for serialization. The downside is StateDelta would miss all maps after `protocol.Decode` call.